### PR TITLE
add http_response plugin

### PR DESCRIPTION
**Pull request by Lswith**
solves issue #332

### CODE REVIEW

**Comment by Lswith** [comment]
I may add the ability to parse http headers in the format:

'''
Host: www.mycompany.com
Content-Type: text
'''

**Comment by Lswith** [comment]
That should be enough input for what I need.

**Comment by sparrc** [comment]
Thanks for the contribution @Lswith, but instead of a standalone plugin, do you think you could add this to the `net_response` plugin?

**Comment by sparrc** [comment]
although the addition of headers may make that difficult

**Comment by Lswith** [comment]
I thought about it and I feel like this is too different to it. Yes they both track response time, however an http request is quite different to a tcp connection.

**Comment by sparrc** [comment]
sure, I think actually it's OK to leave as separate

File: plugins/inputs/http_response/http_response.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  address = "http://github.com"
+  ## Set response_timeout (default 10 seconds)
+  response_timeout = 10
+  ## HTTP Request Method
+  method = "GET"
+  ## HTTP Request Headers
+  headers = '''
+  Host: github.com
+  '''
+	## Whether to follow redirects from the server (defaults to false)
```
Comment (sparrc) [inline plugins/inputs/http_response/http_response.go]: these should be indented by two spaces, not tabs

**Comment by sparrc** [comment]
overall this looks good, but you'll need to write unit tests.

**Comment by Lswith** [comment]
will do.

**Comment by Lswith** [comment]
Tests are complete. Waiting on you guys

File: plugins/inputs/http_response/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+### Configuration:
+
+```
+# List of UDP/TCP connections you want to check
+[[inputs.http_response]]
+  ## Server address (default http://localhost)
+  address = "http://github.com"
+  ## Set response_timeout (default 10 seconds)
+  response_timeout = 10
```
Comment (sparrc) [inline plugins/inputs/http_response/README.md]: default this to 5 seconds

File: plugins/inputs/http_response/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  ## HTTP Request Method
+  method = "GET"
+  ## HTTP Request Headers
```
Comment (sparrc) [inline plugins/inputs/http_response/README.md]: these headers should be specified as a map (see https://github.com/influxdata/telegraf/blob/master/plugins/inputs/httpjson/httpjson.go)

**Comment by sparrc** [comment]
thanks @Lswith, closed by https://github.com/influxdata/telegraf/commit/90185dc6b369981e82ceedc844ce435a000fc094

**Pull request closed without merging**